src/commonMain/kotlin/io/github/kevincianfarini/alchemist/type/Acceleration.kt

```diff
@@ -151,6 +152,22 @@ public value class Acceleration internal constructor(
      */
     public operator fun times(scale: Long): Acceleration = Acceleration(rawNanometersPerSecondSquared * scale)
 
+    /**
+     * Returns an acceleration whose value is multiplied by the specified [scale]. This operation may be rounded when the result
+     * cannot be precisely represented with a [Double] number.
+     *
+     * @throws IllegalArgumentException when this acceleration is [infinite][isInfinite] and [scale] is 0.0 or when this acceleration is 0
+     * and scale is [infinite][Double.isInfinite].
+     */
+    public operator fun times(scale: Double): Acceleration {
+        val intScale = scale.roundToInt()
+        if (intScale.toDouble() == scale) {
+            return times(intScale)
+        } else {
+            return Acceleration(rawNanometersPerSecondSquared * scale)
+        }
```

Comment:
Rather than replicating the `if/else` in each quantity everywhere, I think this can be done in `SaturatingLong.times(Double)`. Sorry if that wasn't clear from my previous comments -- that's what I was trying to convey.